### PR TITLE
Supporting User MSI in CMK

### PR DESCRIPTION
**Pull request by asmaskar**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by j5lim** [comment]
>             "description": "OK"

PUT should return the original request at least. Please see the RPC: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#put-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#put-resource)

---
Refers to: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/bms.json:692 in 129ac3e. [](commit_id = 129ac3ee415cb44dc3d15b337df5cac447240ae0, deletion_comment = False)

**Comment by j5lim** [comment]
>     "BackupResourceEncryptionConfigExtended": {

Why not actually extending BackupResourceEnruptionConfig using 'allof'?

---
Refers to: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/bms.json:3977 in 129ac3e. [](commit_id = 129ac3ee415cb44dc3d15b337df5cac447240ae0, deletion_comment = False)

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/examples/BackupResourceEncryptionConfig_Put.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "parameters": {
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "resourceGroupName": "test-rg",
+    "vaultName": "source-rsv",
+    "api-version": "2021-01-01",
+    "parameters": {
+      "properties": {
```
Comment (j5lim) [inline specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/examples/BackupResourceEncryptionConfig_Put.json:8]: >"properties":  [](http://example.com/codeflow?start=6&length=14)

Can userAssignedIdentity and useSystemAssignedIdentity be set by user? Can we update the example?

**Review by j5lim — CHANGES_REQUESTED** [review]
Left comments.

**Comment by asmaskar** [comment]
Hi @j5lim,
Wrt to the comment regarding the PUT resource response, Yes it should ideally have the original request, but i think making that change would be a breaking change. As this was checked in an year ago may be, and as SDK for this is already released, should we be changing the response for it now?

and about extending the BackupEncryptionConfig, actually this swagger was auto generated by the tool, hence I made the changes accordingly. And also one point is that, we might actually remove the section of BackupEncryptionConfig and keep only BackupEncryptionConfigExtended. So with this, do you think the modification is needed?

**Comment by asmaskar** [comment]
Hi @j5lim ,

I have made a fix in the PR. I made a mistake by changing the Put Request. The Put request is supposed to be the same as in customer is not supposed to set userAssignedIdentity using this API. Although in GET he will receive those information as it is modified using some other API.

File: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/examples/BackupResourceEncryptionConfig_Put.json (hunk 1 above)
Comment (asmaskar) [reply specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/examples/BackupResourceEncryptionConfig_Put.json:8]: Made a fix. User is not supposed to set the fields using this API. The information will only be received as a part of GET call

**Comment by asmaskar** [comment]
with respect to the other comments, the PUT call response was checked in may be more than an year ago, changing that, wont it be a breaking change? as SDK is released , is it ideal to make the change there?

Regarding the other comment of using allof, i generated this swagger using a tool, and actually not much context of how you are suggesting the extending of BackupEncryptionConfig. Can you point me to some place where i can refer for the mentioned changes?

**Comment by j5lim** [comment]
>           "type": "boolean"

Please mark them as read-only since the user is not allowed to set those in PUT.

---
Refers to: specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2021-04-01/bms.json:3991 in 263ee96. [](commit_id = 263ee968d96cb6e9249eab5fc860fce37e028891, deletion_comment = False)

**Comment by j5lim** [comment]
Right, the service code need to be fixed before updating the swagger. Can you please create a WI for that so that it can be fixed in the next release?

---
In reply to: [824402975](https://github.com/Azure/azure-rest-api-specs/pull/14048#issuecomment-824402975) [](http://example.com/codeflow?ancestors=824402975,824402975)

**Review by j5lim — CHANGES_REQUESTED** [review]
Added new comments.

**Comment by asmaskar** [comment]
Hi,
I am not sure whether we should be marking the fields readonly as these fields can be updated using a different API.

**Comment by asmaskar** [comment]
@j5lim Can you please have a check if i answered your queries

**Review by j5lim — APPROVED** [review]
Approved from ARM side.

**Comment by chunyu3** [comment]
@asmaskar There are cross version breaking changes, and it need to acquire breaking change approve.
And there are some 'duplicate schema' issues ( see details in SDK Track2 validation), would you please fix this also? Thanks

**Comment by pratimaupadhyay02** [comment]
> @asmaskar There are cross version breaking changes, and it need to acquire breaking change approve.
> And there are some 'duplicate schema' issues ( see details in SDK Track2 validation), would you please fix this also? Thanks

@chunyu3 We are making fix for the issue "Duplicate Schema named NewErrorResponse -- properties.error.$ref: undefined => "#/components/schemas/schemas:1024"". Seems like the NewErrorResponse schema in our spec and the Error Response under common json have same definition so we are making changes to use the ErrorResponse from common schema as itself. 

Regarding the Duplicated schema issues, I suspect it is due to the fact that in our package, we have a version 2021-03-01 that contains definitions of all APIs(all these APIs are being routed to active stamp), and another version 2018-12-20 in the same package which contains APIs which are Cross Region restore specific which need to be routed to passive stamp based on the API version, So it is expected that the same API definitions are present in both these version hence the validation issue. Since this is expected for other API definitions apart from NewErrorResponse, is there a provision to ignore this issue for the other APIs?

**Comment by asmaskar** [comment]
Hi @chunyu3 , the SDK Track2 validation issue is not in our control, as in its related to some other issue. And as far as the breaking change is concerned the error there is shown due to weaker enum constraint but we havent removed any values from the enum. We have added values. Can you please let us know if there are any other issues in it

**Comment by chunyu3** [comment]
> Hi @chunyu3 , the SDK Track2 validation issue is not in our control, as in its related to some other issue. And as far as the breaking change is concerned the error there is shown due to weaker enum constraint but we havent removed any values from the enum. We have added values. Can you please let us know if there are any other issues in it

Hi @asmaskar For the enum breaking change, yes, it may not be the issue of your PR, you can ignore this. I will confirm with the breakingchange detect owner about it. If there still issue, I will contact you.
For other breaking changes (such as remove path comparing with preview version) (see Cross Version BreakingChange (Base on preview version) ), those breaking change need approve, please take a look at them and follow breaking change approve process. Thanks,

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**